### PR TITLE
Condense restore feed configuration to single location and remove package mirroring

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves https://github.com/aspnet/Universe/issues/856

Changes:
 - Update README to point to the dotnet-core feed, not aspnetcore-dev (aspnet/Announcements#293)
 - Stop mirroring partner team packages to aspnetcore-dev
 - Centralize myget feed configuration to one file
 - Stop downloading mirror packages ahead of time
 - Reorganize external-dependencies.props to be sorted by package ID, not by the feed from which the package comes

### CODE REVIEW

File: build/Publish.targets [hunk 1]
```diff
@@ -195,7 +190,6 @@
     <!-- Contains the super set of packages that we'd like to publish to MyGet and Azure blob -->
     <ItemGroup>
       <PackageToPublishToMyGet Include="@(PackageToPublish)" />
-      <PackageToPublishToMyGet Include="@(PackageToMirror)" />
```
Comment (pranavkm) [inline build/Publish.targets]: 👏

File: build/sources.props [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      $(RestoreSources);
+      https://api.nuget.org/v3/index.json;
+    </RestoreSources>
+    <RestoreSources Condition=" '$(DisableMyGetRestoreSources)' != 'true' ">
+      $(RestoreSources);
+      https://dotnet.myget.org/F/dotnet-core/api/v3/index.json;
+      https://dotnet.myget.org/F/aspnetcore-tools/api/v3/index.json;
+      https://dotnet.myget.org/F/aspnetcore-master/api/v3/index.json;
+      https://dotnet.myget.org/F/roslyn/api/v3/index.json;
+      https://vside.myget.org/F/vssdk/api/v3/index.json;
```
Comment (pranavkm) [inline build/sources.props]: I'll send a Razor PR to start referencing the extra feeds. Could you take care of  the xunit one?

Comment (pranavkm) [reply build/sources.props]: Nevermind. Looks like it's already in place

**Comment by natemcmaster** [comment]
:watch: waiting on prodcon stabilization. This is potentially disruptive, so waiting till next week to merge.

**Pull request merged**